### PR TITLE
Improve Error Handling and Add more Validation

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?

This PR improve the error handling and add additional validation for the service account JSON fields.
## Checklist

- [X] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/config/errors.php [hunk 1]
```diff
@@ -728,4 +728,9 @@
         'description' => 'Migration is already in progress. You can check the status of the migration in your Appwrite Console\'s "Settings" > "Migrations".',
         'code' => 409,
     ],
+    Exception::MIGRATION_PROVIDER_ERROR => [
+        'name' => Exception::MIGRATION_PROVIDER_ERROR,
+        'description' => 'An error occurred on the provider\'s side. Please try again later.',
+        'code' => 424,
```
Comment (christyjacob4) [inline app/config/errors.php:734]: Let's use a 400 error here...

Comment (christyjacob4) [reply app/config/errors.php:734]: Can you also make sure the frontend is handling and displaying it correctly ?

Comment (PineappleIOnic) [reply app/config/errors.php]: Frontend isn't currently displaying this correctly, bringing it up to the frontend team

**Review by christyjacob4 — APPROVED** [review]

**Review by TorstenDittmann — APPROVED** [review]

**Comment by stnguyen90** [comment]
@PineappleIOnic there are still merge conflicts.

**Review by eldadfux — APPROVED** [review]

**Pull request merged**